sandbox/apparmor/prompting/ioctl.go

```diff
@@ -0,0 +1,139 @@
+package prompting
```

Comment:
thinking further, maybe under sandbox/apparmor this should be called notify?

Reply:
The ioctl component, or all of this `prompting` package?

Michael suggested moving `prompting/notifier` to `sandbox/apparmor/listen`, do you think that this `ioctl.go` file should be moved under the current `notifier` package, and that should be renamed to `listen`? Or should `ioctl.go` be moved to its own `notify` package, with the current `notifier` becoming `sandbox/apparmor/listen`?

Reply:
all of this prompting package, I cannot say much about the notifier because I'm a bit unsure what it does

Reply:
The notifier uses ioctl to send and receive messages from apparmor. The current state of the notifier is here (fairly short): https://github.com/snapcore/snapd/blob/058ffd3e791255867619804a71a20de25e7562b9/prompting/notifier/notifier.go

Originally (years ago) the notifier was part of this same package, but it was moved out, since it theoretically didn't need to know about the implementation details of the `apparmor` (now `prompting`) package. In my opinion, a lot of the notifier's use of this package should be abstracted further (there's still some manual struct creation, which I don't love).

Do you think the notifier should be re-integrated into this package, or should ioctl be moved into the notifier, or should `notifier` be renamed to something else?

Reply:
at the moment my thinking would be to have:

this package renamed to sandbox/apparmor/notify

the notifier to live (at least initially) under sandbox/apparmor/notify/listen